src/Foundation/Commerce/code/Services/Billing/BillingService.cs

```diff
@@ -15,28 +15,231 @@
+
+    using FederatedPaymentInfo = Models.Entities.Payment.FederatedPaymentInfo;
+    using PaymentOption = Models.Entities.Payment.PaymentOption;
+    using PaymentMethod = Models.Entities.Payment.PaymentMethod;
 
     [Service(typeof(IBillingService), Lifetime = Lifetime.Singleton)]
     public class BillingService : IBillingService
     {
+        private readonly ICartManagerV2 cartManager;
+        private readonly IPaymentMapper paymentMapper;
+        private readonly IPaymentManagerV2 paymentManager;
+        private readonly IStorefrontContext storefrontContext;
+        private readonly IVisitorContext visitorContext;
+
+        public BillingService(
+            ICartManagerV2 cartManager,
+            IStorefrontContext storefrontContext,
+            IVisitorContext visitorContext,
+            IPaymentMapper paymentMapper,
+            IPaymentManagerV2 paymentManager)
+        {
+            Assert.ArgumentNotNull(cartManager, nameof(cartManager));
+            Assert.ArgumentNotNull(paymentMapper, nameof(paymentMapper));
+            Assert.ArgumentNotNull(paymentManager, nameof(paymentManager));
+            Assert.ArgumentNotNull(storefrontContext, nameof(storefrontContext));
+            Assert.ArgumentNotNull(visitorContext, nameof(visitorContext));
+
+            this.cartManager = cartManager;
+            this.paymentMapper = paymentMapper;
+            this.storefrontContext = storefrontContext;
+            this.visitorContext = visitorContext;
+            this.paymentManager = paymentManager;
+        }
+
         public Result<BillingInfo> GetBillingOptions()
         {
-            throw new NotImplementedException();
+            var model = new BillingInfo();
+            var result = new Result<BillingInfo>(model);
+
+            if (!Sitecore.Context.PageMode.IsExperienceEditor)
+            {
+                result.SetResult(model);
+                var cartResult = this.cartManager.LoadCart(
+                    this.storefrontContext.ShopName,
+                    this.visitorContext.ContactId);
+
+                if (!cartResult.Success)
+                {
+                    result.SetErrors(cartResult.SystemMessages.Select(m => m.Message).ToList());
+
+                    return result;
+                }
+
+                if (cartResult.Cart.Lines != null && cartResult.Cart.Lines.Any())
+                {
+                    this.AddPaymentOptions(result, cartResult.Cart);
+
+                    if (result.Success)
+                    {
+                        this.AddPaymentMethods(result, cartResult.Cart);
+
+                        if (result.Success)
+                        {
+                            this.AddPaymentClientToken(result);
+                        }
+                    }
+                }
+            }
+
+            return result;
         }
 
         public Result<VoidResult> SetPaymentOptions(Address billingAddress, FederatedPaymentInfo federatedPayment)
```

Comment:
This method is pretty huge and do several things, not only `SetPaymentOptions` but:
1. Updates cart email
2. Adds addresses to cart
3. Adds payment info to cart